packages/apps/tests/test_streaming.py

```diff
@@ -0,0 +1,161 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+# pyright: basic
+
+import asyncio
+from datetime import datetime, timedelta
+from unittest.mock import MagicMock
+
+import pytest
+from microsoft.teams.api import (
+    Account,
+    ApiClient,
+    ConversationAccount,
+    ConversationReference,
+    SentActivity,
+    TypingActivityInput,
+)
+from microsoft.teams.apps import HttpStream
+
+
+class TestHttpStream:
+    @pytest.fixture
+    def mock_logger(self):
+        return MagicMock()
+
+    @pytest.fixture
+    def mock_api_client(self):
+        client = MagicMock(spec=ApiClient)
+
+        mock_activities = MagicMock()
+        mock_conversations = MagicMock()
+        mock_conversations.activities.return_value = mock_activities
+        client.conversations = mock_conversations
+
+        client.send_call_count = 0
+        client.send_times = []
+        client.sent_activities = []
+
+        async def mock_send(activity):
+            client.send_call_count += 1
+            client.send_times.append(datetime.now())
+            client.sent_activities.append(activity)
+            return SentActivity(id=f"test-id-{client.send_call_count}", activity_params=activity)
+
+        client.conversations.activities().create = mock_send
+
+        return client
+
+    @pytest.fixture
+    def conversation_reference(self):
+        return ConversationReference(
+            service_url="https://smba.trafficmanager.net/teams/",
+            bot=Account(id="test-bot", name="Test Bot"),
+            conversation=ConversationAccount(id="test-conversation", conversation_type="personal"),
+            activity_id="test-activity",
+            channel_id="msteams",
+        )
+
+    @pytest.fixture
+    def http_stream(self, mock_api_client, conversation_reference, mock_logger):
+        return HttpStream(mock_api_client, conversation_reference, mock_logger)
+
+    @pytest.mark.asyncio
+    async def test_stream_emit_message_flushes_after_500ms(self, mock_api_client, conversation_reference, mock_logger):
```

Comment:
can we add into one of the test cases to check a sequence of `.update` and `.emit`